app/controllers/HealthDataController.java

```diff
@@ -34,6 +34,22 @@ public Result createRecord() {
     }
 
     /**
+     * Play controller for DELETE /admin/v2/healthdata/byHealthCode/:healthCode. This API is used by admins to delete
+     * health data records for a given health code (user in study).
+     *
+     * @param healthCode
+     *         health code to delete records for
+     * @return Play result indicating number of records deleted
+     */
+    public Result deleteRecordsForHealthCode(String healthCode) {
```

Comment:
Nothing in the API should ever expose the health code such that it could be used to delete records through the API, and the only place we want to do this is when we're deleting a user, so I think we should remove the routes and controller code. Note the user has the alternative to turn sharing of these records off through the api, and that preference is written to the ParticipantOptions table. When exporting, any health codes mapped to a preference not to share, should have their data records excluded from the export. But that's all that is exposed.